PlaybackLab/FullscreenSampleApp/app/src/main/java/com/ooyala/fullscreensampleapp/PlayerActivity.java

```diff
@@ -35,20 +36,33 @@
     private List<Data> dataList;
 
     private class ScrollListener extends RecyclerView.OnScrollListener {
-        private int snapPosition = RecyclerView.NO_POSITION;
-
-        @Override
-        public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newState) {
-            super.onScrollStateChanged(recyclerView, newState);
+        private static final int PLAY_DELAY = 500;
 
-            if (newState == RecyclerView.SCROLL_STATE_IDLE) {
+        private int snapPosition = RecyclerView.NO_POSITION;
+        private int state = RecyclerView.SCROLL_STATE_IDLE;
+        private Handler handler = new Handler();
+        private Runnable runnable = () -> {
```

Comment:
What do you think about extracting this initialization in method?

Reply:
I don't like it because there are lots of play methods here. I don't want to add smth like `initAndPlay()`.

Reply:
maybe we can just rename 'runnable' ? smth like
Runnable initAndPlayRunnable  = () -> { ... }

Reply:
I'll change it on `playRunnable` because this is the final result of the `runnable `

Reply:
done